### ISSUE TITLE
Pydantic v1 namespace compatibility

### ISSUE THREAD

**Issue by josephgruber-umbra** [developmentseed/geojson-pydantic#166]
Can a v0.7 be released that switches to using the Pydantic v1 namespace? This would unblock a migration path to Pydantic v2 for those who can't immediately migrate to Pydantic v2 features/capabilities.

More details on the v1 namespace at https://docs.pydantic.dev/latest/migration/#continue-using-pydantic-v1-features

**Comment by vincentsarago** [comment]
@josephgruber-umbra I don't have bandwidth to work on this at the moment but I'll be happy to review a PR 🙏

**Comment by josephgruber-umbra** [comment]
@vincentsarago Here is a commit I'd like to create a PR for: https://github.com/developmentseed/geojson-pydantic/compare/main...josephgruber-umbra:geojson-pydantic:pydantic_v1_namespace_support. I have not created the PR as this should not be merged into `main` (nor any of the other branches currently available). My recommendation is a 0.7.0 version/tag branched from the 0.6.3 version/tag. 

The goal with these changes is to continue using Pydantic v1, via the v1 namespace that was added in Pydantic 1.10.17 a few weeks ago, while opening up the migration path to Pydantic v2 which also includes the v1 namespace. More details from Pydantic on the v1 namespace and operating in a v1/v2 environment can be found at https://docs.pydantic.dev/latest/migration/#using-pydantic-v1-features-in-a-v1v2-environment.

Ideally with semver this would negate a major version change however with 1.x already supporting (only) Pydantic v2, I believe a 0.7.0 approach works since this does not force a Pydantic v2 upgrade nor does it change any functionality from Pydantic v1, just the Python module namespace. It would not be expected any future updates would be made outside of this change, nor any changes after 0.6.3 backported. Simply a migration-friendly version.

**Comment by vincentsarago** [comment]
@josephgruber-umbra sorry I've totally missed this comment :-( 

Feel free to create a PR on top of this branch https://github.com/developmentseed/geojson-pydantic/tree/maint-0.x

**Comment by vincentsarago** [comment]
thanks @josephgruber-umbra 

0.7 is now on pypi https://pypi.org/project/geojson-pydantic/0.7.0/ 🙏
